### PR TITLE
[Wav2Vec2] Rename model's feature extractor to feature encoder

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

In this PR all `<SpeechModel>FeatureExtractor` are renamed to `<SpeechModel>FeatureEncoder` and `--freeze_feature_extractor` is renamed to `--freeze_feature_encoder` 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/visual_bert/modeling_visual_bert.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def __init__(self, config):
         # position_ids (1, len position emb) is contiguous in memory and exported when serialized
         self.register_buffer("position_ids", torch.arange(config.max_position_embeddings).expand((1, -1)))
 
-        # For Visual Features
+        # For Visual FeatureEncoder
```
Comment (patrickvonplaten) [inline src/transformers/models/visual_bert/modeling_visual_bert.py:87]:
```suggestion
        # For Visual Features
```

File: src/transformers/models/visual_bert/modeling_visual_bert.py [hunk 2]
```diff
@@ -1571,7 +1571,7 @@ def forward(
         )
         selected_positions = sequence_output.gather(1, expanded_region_to_phrase_positions)
 
-        # Visual Features = batch x visual_feature_length x dim
+        # Visual FeatureEncoder = batch x visual_feature_length x dim
```
Comment (patrickvonplaten) [inline src/transformers/models/visual_bert/modeling_visual_bert.py:1574]:
```suggestion
        # Visual Features = batch x visual_feature_length x dim
```

File: src/transformers/models/unispeech_sat/convert_unispeech_sat_original_pytorch_checkpoint_to_pytorch.py [hunk 3]
```diff
@@ -20,7 +20,7 @@
 import fairseq
 import torch
 
-from transformers import (  # UniSpeechSatCTCTokenizer,; UniSpeechSatFeatureExtractor,; UniSpeechSatProcessor,
+from transformers import (  # UniSpeechSatCTCTokenizer,; UniSpeechSatFeatureEncoder,; UniSpeechSatProcessor,
```
Comment (patrickvonplaten) [inline src/transformers/models/unispeech_sat/convert_unispeech_sat_original_pytorch_checkpoint_to_pytorch.py:23]:
```suggestion
from transformers import (
```

File: src/transformers/models/hubert/modeling_hubert.py [hunk 4]
```diff
@@ -1063,8 +1064,20 @@ def __init__(self, config):
...
-        will not be updated during training.
+        Calling this function will disable the gradient computation for the feature encoder so that its parameter will
+        not be updated during training.
+        """
+        warnings.warn(
+            "The method `freeze_feature_extractor` is deprecated and will be removed in Transformers v5."
+            "Please use the equivalent `freeze_feature_encoder` method instead.",
+            FutureWarning,
+        )
+        self.hubert.feature_extractor._freeze_parameters()
```
Comment (sgugger) [inline src/transformers/models/hubert/modeling_hubert.py:1075]: Good practice is to call the method that will replace it, so if you change code there for some reason, the code is automatically up to date.

Putting it just here once, but it's valid for all other models.

**Review by sgugger — COMMENTED** [review]
Thanks a lot for doing the switch.

The PR is slightly breaking without a typealias from the old `ModelTypeFeatureExtractor` to the new `ModelTypeFeatureEncoder` class, but this is tiny and for an internal class of the modeling file, so I think this is acceptable for recent models (I wouldn't say the same thing for Bert for instance!)

File: src/transformers/models/hubert/modeling_hubert.py (hunk 4 above)
Comment (patrickvonplaten) [reply src/transformers/models/hubert/modeling_hubert.py:1075]: > Thanks a lot for doing the switch.
> 
> The PR is slightly breaking without a typealias from the old `ModelTypeFeatureExtractor` to the new `ModelTypeFeatureEncoder` class, but this is tiny and for an internal class of the modeling file, so I think this is acceptable for recent models (I wouldn't say the same thing for Bert for instance!)

Yeah good point - will actually leave an alias

File: src/transformers/models/hubert/modeling_tf_hubert.py [hunk 5]
```diff
@@ -659,7 +659,7 @@ def call(self, hidden_states):
         return hidden_states
 
 
-class TFHubertFeatureExtractor(tf.keras.layers.Layer):
+class TFHubertFeatureEncoder(tf.keras.layers.Layer):
```
Comment (sgugger) [inline src/transformers/models/hubert/modeling_tf_hubert.py:662]: You didn't put the typealias in this file. Is it on purpose?

File: src/transformers/models/wav2vec2/modeling_tf_wav2vec2.py [hunk 6]
```diff
@@ -655,7 +655,7 @@ def call(self, hidden_states):
         return hidden_states
 
 
-class TFWav2Vec2FeatureExtractor(tf.keras.layers.Layer):
+class TFWav2Vec2FeatureEncoder(tf.keras.layers.Layer):
```
Comment (sgugger) [inline src/transformers/models/wav2vec2/modeling_tf_wav2vec2.py:658]: Same here.

**Review by sgugger — APPROVED** [review]

**Pull request merged**